### PR TITLE
Adds master folio fetch endpoint.

### PR DESCRIPTION
**Pull request by laxderek**
[#145300617]

https://www.pivotaltracker.com/story/show/145300617

### CODE REVIEW

**Comment by laxderek** [comment]
@pcsouza PTAL

File: proto/money/rpc/master_folio_svc.proto [hunk 1]
```diff
@@ -48,10 +49,23 @@ enum MasterFolioSvcFolioSwapResult {
...
+	.holms.types.booking.indicators.ReservationIndicator client_folio = 1;
+	.holms.types.booking.indicators.ReservationIndicator master_folio = 2;
+}
+
 message MasterFolioSvcFolioSwapResponse {
 	MasterFolioSvcFolioSwapResult result = 1;
 }
 
+message MasterFolioSvcGetMasterFolioAssignmentsRequest {
+	.holms.types.booking.reservations.ReservationQueryFilter filter = 1;
```
Comment (pcsouza) [inline proto/money/rpc/master_folio_svc.proto]: This isn't the correct search parameter. It should be a repeated field of reservation indicators:

1) This isn't the booking context, so we shouldn't be porting booking context concepts into here
2) The reservations display has changed significantly so there are a multitude of ways reservations get queried, not to mention the reservation search.

File: proto/money/rpc/master_folio_svc.proto [hunk 2]
```diff
@@ -48,10 +49,23 @@ enum MasterFolioSvcFolioSwapResult {
 	MASTER_FOLIO_SVC_FOLIO_SWAP_FAILURE_UNKNOWN = 3;
 }
 
+message MasterFolioSvcReservationMasterFolioAssignmentList {
+	repeated ReservationMasterFolioAssignment assignments = 1;
+}
+
+message ReservationMasterFolioAssignment {
+	.holms.types.booking.indicators.ReservationIndicator client_folio = 1;
```
Comment (pcsouza) [inline proto/money/rpc/master_folio_svc.proto]: This needs to be a repeated field. A master can have multiple clients. As for displaying that fact...I think we need to talk to Andrew.

**Comment by laxderek** [comment]
@pcsouza PTAL

**Pull request merged**